src/renderer/app.jsx

```diff
@@ -38,9 +43,32 @@ const ScratchDesktopHOC = function (WrappedComponent) {
                 'handleTelemetryModalOptIn',
                 'handleTelemetryModalOptOut'
             ]);
+            this.state = {
+                // projectFileName is used as "do we want to wait for a project file to load?"
+                projectFileName: mainProcessArgs.projectFile,
+                projectData: null
+            };
         }
         componentDidMount () {
             ipcRenderer.on('setTitleFromSave', this.handleSetTitleFromSave);
+            if (this.state.projectFileName) {
+                fs.readFile(this.state.projectFileName, null, (err, data) => {
+                    if (err) {
+                        remote.dialog.showMessageBox({
+                            type: 'error',
+                            title: 'Failed to load project',
+                            message: `Could not load project from file:\n${this.state.projectFileName}`,
+                            detail: err.message,
```

Comment:
This `err.message` ends up looking like the image below, which is a little more technical than would be ideal.

![image](https://user-images.githubusercontent.com/7019101/68801219-7a573380-0610-11ea-946d-0ccda898f114.png)